pkg/provider/identity.go

```diff
@@ -163,7 +171,7 @@ func PeerIdentityFromPeer(peer *peer.Peer) (*PeerIdentity, error) {
 	if len(c) < 2 {
 		return nil, Error.New("invalid certificate chain")
 	}
-	pi, err := PeerIdentityFromCerts(c[0], c[1])
+	pi, err := PeerIdentityFromCerts(c[0], c[1], c[2:])
```

Comment:
nit -- I wonder if it would make since to abstract away PeerCertificates and VerifiedChains so we can add methods on them vs indexing. No changes needed just a random thought